src/libecalc/presentation/yaml/yaml_types/emitters/yaml_venting_emitter.py

```diff
@@ -1,36 +1,139 @@
 from libecalc.presentation.yaml.yaml_types.components.yaml_category_field import (
     CategoryField,
 )
 from libecalc.presentation.yaml.yaml_types.yaml_temporal_model import YamlTemporalModel
+from libecalc.presentation.yaml.yaml_types.yaml_to_core_base_models import (
+    YamlBaseEquipment,
+    YamlEcalcBaseModel,
+)
+from libecalc.presentation.yaml.yaml_types.yaml_variable import YamlDefaultDatetime
 
 
-class YamlEmitterModel(YamlBase):
+class YamlEmitterModel(YamlEcalcBaseModel):
+    name: Optional[ComponentNameStr] = Field("", title="NAME", description="Name of emitter model")
+    user_defined_category: Optional[str] = CategoryField(
+        "",
+    )
     emission_rate: ExpressionType = Field(
         ...,
         title="EMISSION_RATE",
         description="Emission rate expression [kg/day]",
     )
-
-
-class YamlVentingEmitter(YamlBase):
-    name: str = Field(
+    emission_rate_type: RateType = Field(
         ...,
-        title="NAME",
-        description="Name of the emitter.\n\n$ECALC_DOCS_KEYWORDS_URL/NAME",
+        title="EMISSION_RATE_TYPE",
+        description="Emission rate type, calendar day or stream day",
     )
-    category: YamlTemporalModel[ConsumerUserDefinedCategoryType] = CategoryField(
+    regularity: Dict[YamlDefaultDatetime, Expression] = Field(
         ...,
+        title="REGULARITY",
+        description="Regularity",
     )
-    emission_name: str = Field(
+
+    _validate_regularity_temporal_model = validator("regularity", allow_reuse=True)(validate_temporal_model)
+    _default_emission_rate = validator("emission_rate", allow_reuse=True)(convert_expression)
+
+
+class YamlTemporalEmitterModel:
+    def __init__(self, target_period: Period):
+        self._target_period = target_period
+
+    @staticmethod
+    def create_model(model: Dict, regularity: Dict[YamlDefaultDatetime, Expression]):
+        emission_rate = model.get(EcalcYamlKeywords.installation_venting_emitter_emission_rate)
+        emission_rate_type = model.get(EcalcYamlKeywords.venting_emitter_rate_type) or RateType.STREAM_DAY
+
+        try:
+            return YamlEmitterModel(
+                emission_rate=emission_rate,
+                regularity=regularity,
+                emission_rate_type=emission_rate_type,
+            )
+        except ValidationError as e:
+            raise ValueError(e) from e  # Got circular import issues with DtoValidationError...
+
+    def get_model(
+        self, data: Optional[Dict], regularity: Dict[YamlDefaultDatetime, Expression]
+    ) -> Dict[YamlDefaultDatetime, YamlEmitterModel]:
+        time_adjusted_model = define_time_model_for_period(data, target_period=self._target_period)
+        return {
+            YamlDefaultDatetime(
+                start_date.year, start_date.month, start_date.day, start_date.hour, start_date.second
+            ): self.create_model(model, regularity=regularity)
+            for start_date, model in time_adjusted_model.items()
+        }
```

Comment:
Do we need this?

Reply:
The `YamlTemporalEmitterModel`, the full code-block was what I meant

Reply:
Will have a look.